mobile/cmp/navigator/NavigatorModel.ts

```diff
@@ -176,85 +266,56 @@ export class NavigatorModel extends HoistModel {
                 return;
             }
 
-            const page = new PageModel({
-                navigatorModel: this,
-                ...mergeDeep({}, pageModelCfg, part)
-            });
-
-            stack.push(page);
+            // Re-use existing PageModels where possible
+            const existingPageModel = i < this.stack.length ? this.stack[i] : null;
```

Comment:
this could "simplify" to just this.stack[i], with out-of-bound behavior for javascript arrays